.github/workflows/build.yml

```diff
@@ -0,0 +1,82 @@
+name: Build k3s
+on:
+  schedule:
+    - cron: '0 0 * * *'
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'The version of K3s to build an image for e.g. v1.28.3'
+        required: true 
+        type: string
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Get K3s version
+        id: version
+        run: |-
+          # Get latest K3s version
+          K3S_VERSION=${{inputs.version}}
+          if [ -z "$K3S_VERSION" ]; then
+            K3S_VERSION=$(curl -s https://api.github.com/repos/k3s-io/k3s/releases | \
+              jq -r '.[].tag_name | select(contains("-rc") | not)' | \
+              sort -V -r | \
+              head -n 1)
+            CURRENT_VERSION=$(curl --silent --fail -L \
```

Comment:
If we create a tag for every build, shouldn't we just check whether the tag for `$K3S_VERSION` already exists?